src/test/custom_node_buffersqueue_test.cpp

```diff
@@ -0,0 +1,142 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <algorithm>
+#include <cstring>
+#include <vector>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "../custom_nodes/common/buffersqueue.hpp"
+
+using namespace ovms;
+using custom_nodes_common::BuffersQueue;
+
+TEST(CustomNodeBuffersQueue, GetAllBuffers) {
+    const std::string content{"abc"};
+    size_t buffersCount = 12;
+    BuffersQueue buffersQueue(content.size(), buffersCount);
+    std::vector<void*> buffers(buffersCount);
+    for (size_t i = 0; i < buffersCount; ++i) {
+        buffers[i] = buffersQueue.getBuffer();
+        ASSERT_NE(nullptr, buffers[i]) << "Failed to get: " << i;
+        std::memcpy(buffers[i], content.c_str(), content.size());
+    }
+    for (size_t i = 0; i < buffersCount; ++i) {
```

Comment:
This does test content of buffers, but we do not do anything with the buffer other than writing to it.
Maybe it would be better to get buffer, write to it, return buffer, get it again and check the content?

Reply:
When you return the buffer there is no point in reading its content as its out of your control.

Idea here is to write to each of the buffers check if the content is correct. Then sort all addresses & check if each buffer size is > content size. With those 2 checks (& sort) we know that no information is lost since there is no overwriting. Its up to buffers queue how internally the buffers addresses are arranged - hence the sort.